### PR TITLE
Add as_dataframe to EffectCollection

### PR DESCRIPTION
**Pull request by arahuja**
This adds an `as_dataframe` to EffectCollection - happy to add or rename any columns.

Also, a few helper properties to `Variant` that I have been using:
- `is_snv`
- `is_transition`
- `is_transversion`

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/varcode/150)

### CODE REVIEW

**Comment by iskandr** [comment]
Very slight nitpick: what do you think of `to_dataframe` instead of `as_dataframe`? It feels slightly more "pythonic" and seems to be the preferred method name for other projects: [BigQuery](http://googlecloudplatform.github.io/datalab/gcp.bigquery.html#gcp.bigquery.Query.to_dataframe), [django-pandas](https://pypi.python.org/pypi/django-pandas), [xarray](http://xarray.pydata.org/en/v0.7.1/generated/xarray.Dataset.to_dataframe.html?highlight=to_dataframe).

**Comment by arahuja** [comment]
Ah yea, I like `to_dataframe` as well, but wanted to match the issue #150

**Comment by tavinathanson** [comment]
Issue https://github.com/hammerlab/varcode/issues/128?

**Comment by arahuja** [comment]
Yes, updated, I meant #128

File: varcode/effect_collection.py [hunk 1]
```diff
@@ -262,3 +263,26 @@ def gene_counts(self):
...
+            row['alt'] = effect.variant.alt
+            row['gene_id'] = effect.gene_id
+            row['gene_name'] = effect.gene_name
+            row['transcript_id'] = effect.transcript_id
+            row['transcript_name'] = effect.transcript_name
+            row['variant'] = str(effect.variant)
+            row['is_indel'] = effect.variant.is_indel
+            row['is_transversion'] = effect.variant.is_transversion
+            row['is_transition'] = effect.variant.is_transition
+            row['is_transversion'] = effect.variant.is_transversion
```
Comment (iskandr) [inline varcode/effect_collection.py]: Twice?

File: varcode/effect_collection.py [hunk 2]
```diff
@@ -262,3 +263,26 @@ def gene_counts(self):
...
+            row['chr'] = effect.variant.contig
+            row['position'] = effect.variant.start
+            row['ref'] = effect.variant.ref
+            row['alt'] = effect.variant.alt
+            row['gene_id'] = effect.gene_id
+            row['gene_name'] = effect.gene_name
+            row['transcript_id'] = effect.transcript_id
+            row['transcript_name'] = effect.transcript_name
+            row['variant'] = str(effect.variant)
+            row['is_indel'] = effect.variant.is_indel
```
Comment (iskandr) [inline varcode/effect_collection.py]: Worth adding 'is_snv' alongside 'is_indel?

File: varcode/effect_collection.py (hunk 1 above)
Comment (arahuja) [inline varcode/effect_collection.py]: Woops, fixed

**Comment by arahuja** [comment]
@iskandr seem to be running into the problem you have on pyensembl, where the py 2.7 tests on Travis on failing with ```The command "nosetests test --with-coverage --cover-package=varcode && ./lint.sh" exited with 137.```

**Comment by iskandr** [comment]
@arahuja Should be fixed now but have to wait for Travis to run. The problem was that I was decoding byte strings into ASCII, which does the sane thing under Python 3 but under Python 3 switches from `str` to a fatter `unicode` representation which uses 2 (or 4!) times as much space per character. The big dictionaries of transcript sequences are about ~200-300mb each per release, so with 4x larger characters it's easy to imagine how we blow through Travis's 3gb limit.

**Comment by arahuja** [comment]
@iskandr Looks to be passing now

**Comment by arahuja** [comment]
@iskandr Good to merge?

**Comment by iskandr** [comment]
@arahuja Definitely

File: varcode/effect_collection.py [hunk 3]
```diff
@@ -262,3 +263,26 @@ def gene_counts(self):
         for effect in self:
             counter[effect.gene_name] += 1
         return counter
+
+    def to_dataframe(self):
+        """Build a dataframe from the effect collection"""
+        def row_from_effect(effect):
+            row = {}
+            row['chr'] = effect.variant.contig
+            row['position'] = effect.variant.start
```
Comment (timodonnell) [inline varcode/effect_collection.py]: might make sense to be clear here about the coordinate system, e.g. call it `inclusive_start`. Just a suggestion, feel free to ignore

File: varcode/effect_collection.py [hunk 4]
```diff
@@ -262,3 +263,26 @@ def gene_counts(self):
         for effect in self:
             counter[effect.gene_name] += 1
         return counter
+
+    def to_dataframe(self):
+        """Build a dataframe from the effect collection"""
+        def row_from_effect(effect):
+            row = {}
+            row['chr'] = effect.variant.contig
```
Comment (timodonnell) [inline varcode/effect_collection.py]: I think `contig` might be better than `chr`. Also, I'd suggest including the reference genome in each row (e.g. a column called `genome` that gets set to `grch37` in every row)

**Comment by timodonnell** [comment]
these are the column names i've been using. Not necessary to match them but just fwiw https://github.com/hammerlab/varlens/blob/master/varlens/variants_util.py#L26

File: varcode/effect_collection.py (hunk 4 above)
Comment (arahuja) [inline varcode/effect_collection.py]: Yea, I think it's best to match the varcode field names

File: varcode/effect_collection.py (hunk 3 above)
Comment (arahuja) [inline varcode/effect_collection.py]: Similarly, I'll make it match the varcode field names

**Pull request merged**